### PR TITLE
chore: add dependabot and dev-container spec

### PR DESCRIPTION
**Pull request by rokroskar**
This PR adds a configuration for dependabot updates as well as a dev-container spec that allows development with "batteries included" either locally or in a GitHub Codespace.

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+# [Choice] Python version (use -bullseye variants on local arm64/Apple Silicon): 3, 3.10, 3.9, 3.8, 3.7, 3.6, 3-bullseye, 3.10-bullseye, 3.9-bullseye, 3.8-bullseye, 3.7-bullseye, 3.6-bullseye, 3-buster, 3.10-buster, 3.9-buster, 3.8-buster, 3.7-buster, 3.6-buster
+ARG VARIANT="3.10-bullseye"
+FROM mcr.microsoft.com/vscode/devcontainers/python:0-${VARIANT}
+
+# [Choice] Node.js version: none, lts/*, 16, 14, 12, 10
+ARG NODE_VERSION="none"
+RUN if [ "${NODE_VERSION}" != "none" ]; then su vscode -c "umask 0002 && . /usr/local/share/nvm/nvm.sh && nvm install ${NODE_VERSION} 2>&1"; fi
+
+# [Optional] If your pip requirements rarely change, uncomment this section to add them to the image.
```
Comment (Panaetius) [inline .devcontainer/Dockerfile:11]: but it's already uncommented :wink:

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+COPY requirements.txt /tmp/pip-tmp/
+RUN pip3 --disable-pip-version-check --no-cache-dir install -r /tmp/pip-tmp/requirements.txt \
+   && rm -rf /tmp/pip-tmp
+
+# [Optional] Uncomment this section to install additional OS packages.
+RUN apt-get update -y \
```
Comment (Panaetius) [inline .devcontainer/Dockerfile:17]: this is also already uncommented

**Review by Panaetius — COMMENTED** [review]
You might want to add https://github.com/SwissDataScienceCenter/renku-python/blob/develop/.github/workflows/combine-dependabot-updates.yml without the poetry merge driver section. it allows to group all dependency updates into a single PR by clicking the `Run Action` button e.g. here: https://github.com/SwissDataScienceCenter/renku-python/actions/workflows/combine-dependabot-updates.yml

**Review by Panaetius — APPROVED** [review]

**Comment by rokroskar** [comment]
Thanks for the suggestion about combining the PRs - I think I'll leave it as-is for now and pull that info out if it becomes unbearable. I limited the number of simultaneous PRs to 5 so hopefully that will be manageable.

**Review by Panaetius — APPROVED** [review]

**Pull request merged**